### PR TITLE
Add Pseudo-column for fileSize and fileModifiedTime

### LINKED ISSUE #1423: Pseudocolumn for "last modified" date
The $path pseudo column is very helpful; Thank you for providing this.
Could the metadata for the "last modified" date be added as well?

This would allow me to:
- Selectively pull 'new' data from an object path (IE ETL processed up to X date, get 'new' data )
- Summarize S3 details directly in a query without relying on external processes 

My current usage of Presto is with Athena.
Thank you!

### PR DESCRIPTION
**Pull request by Praveen2112**
Fixes #1423. Added two hidden columns for Hive and Iceberg connector.

### CODE REVIEW

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveColumnHandle.java [hunk 1]
```diff
@@ -45,6 +45,11 @@
     public static final HiveType BUCKET_HIVE_TYPE = HIVE_INT;
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveColumnHandle.java]: WRT to commit message:

```
Add Hive Connector $file_size hidden column
```

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 2]
```diff
@@ -2449,6 +2450,59 @@ public void testBucketHiddenColumn()
...
+                assertEquals(fileSizeMap.get(col1).longValue(), fileSize);
+            }
+            else {
+                fileSizeMap.put(col1, fileSize);
+            }
+        }
+        assertEquals(fileSizeMap.size(), 3);
+
+        assertUpdate("DROP TABLE test_file_size");
+        assertFalse(getQueryRunner().tableExists(getSession(), "test_file_size"));
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: not needed assertion

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 3]
```diff
@@ -2449,6 +2450,59 @@ public void testBucketHiddenColumn()
...
+        TableMetadata tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, "test_file_size");
+
+        List<String> columnNames = ImmutableList.of("col0", "col1", PATH_COLUMN_NAME, FILE_SIZE_COLUMN_NAME);
+        List<ColumnMetadata> columnMetadatas = tableMetadata.getColumns();
+        assertEquals(columnMetadatas.size(), columnNames.size());
+        for (int i = 0; i < columnMetadatas.size(); i++) {
+            ColumnMetadata columnMetadata = columnMetadatas.get(i);
+            assertEquals(columnMetadata.getName(), columnNames.get(i));
+            if (columnMetadata.getName().equals(FILE_SIZE_COLUMN_NAME)) {
+                // $file_size should be hidden column
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: not needed comment

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveColumnHandle.java [hunk 4]
```diff
@@ -50,6 +50,11 @@
     public static final HiveType FILE_SIZE_TYPE = HIVE_LONG;
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveColumnHandle.java]: same comment for commit message

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 5]
```diff
@@ -2503,6 +2505,59 @@ public void testFileSizeHiddenColumn()
...
+        TableMetadata tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, "test_file_modified_time");
+
+        List<String> columnNames = ImmutableList.of("col0", "col1", PATH_COLUMN_NAME, FILE_SIZE_COLUMN_NAME, FILE_MODIFIED_TIME_COLUMN_NAME);
+        List<ColumnMetadata> columnMetadatas = tableMetadata.getColumns();
+        assertEquals(columnMetadatas.size(), columnNames.size());
+        for (int i = 0; i < columnMetadatas.size(); i++) {
+            ColumnMetadata columnMetadata = columnMetadatas.get(i);
+            assertEquals(columnMetadata.getName(), columnNames.get(i));
+            if (columnMetadata.getName().equals(FILE_MODIFIED_TIME_COLUMN_NAME)) {
+                // $file_size should be hidden column
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: not needed and wrong comment

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 6]
```diff
@@ -2503,6 +2505,59 @@ public void testFileSizeHiddenColumn()
...
+                assertEquals(fileModifiedTimeMap.get(col1).longValue(), fileModifiedTime);
+            }
+            else {
+                fileModifiedTimeMap.put(col1, fileModifiedTime);
+            }
+        }
+        assertEquals(fileModifiedTimeMap.size(), 3);
+
+        assertUpdate("DROP TABLE test_file_modified_time");
+        assertFalse(getQueryRunner().tableExists(getSession(), "test_file_modified_time"));
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: not needed assertion

**Comment by Praveen2112** [comment]
@kokosing Thanks for your insights. Have applied your comments.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 7]
```diff
@@ -2501,6 +2503,57 @@ public void testFileSizeHiddenColumn()
...
+            int col1 = (int) row.getField(1);
+            long fileModifiedTime = (Long) row.getField(2);
+
+            assertTrue(fileModifiedTime < Instant.now().toEpochMilli());
+            assertEquals(col0 % 3, col1);
+            if (fileModifiedTimeMap.containsKey(col1)) {
+                assertEquals(fileModifiedTimeMap.get(col1).longValue(), fileModifiedTime);
+            }
+            else {
+                fileModifiedTimeMap.put(col1, fileModifiedTime);
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: can we assert that modified time is no earlier than test start time?

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 8]
```diff
@@ -2449,6 +2450,57 @@ public void testBucketHiddenColumn()
...
+            int col1 = (int) row.getField(1);
+            long fileSize = (Long) row.getField(2);
+
+            assertTrue(fileSize > 0);
+            assertEquals(col0 % 3, col1);
+            if (fileSizeMap.containsKey(col1)) {
+                assertEquals(fileSizeMap.get(col1).longValue(), fileSize);
+            }
+            else {
+                fileSizeMap.put(col1, fileSize);
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: can we assert that `fileSize` is greater than `0`?

Comment (Praveen2112) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: We have added that in Line#2490

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java (hunk 2 above)
Comment (kokosing) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: bump

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java (hunk 3 above)
Comment (kokosing) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: bump

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java (hunk 5 above)
Comment (kokosing) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: bump

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java (hunk 6 above)
Comment (kokosing) [reply presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: bump

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveColumnHandle.java (hunk 1 above)
Comment (Praveen2112) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveColumnHandle.java]: Resolved it

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 9]
```diff
@@ -2501,6 +2503,62 @@ public void testFileSizeHiddenColumn()
         assertUpdate("DROP TABLE test_file_size");
     }
 
+    @Test
+    public void testFileModifiedTimeHiddenColumn()
+            throws InterruptedException
+    {
+        long testStartTime = Instant.now().toEpochMilli();
+
+        Thread.sleep(2_000);
```
Comment (kokosing) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java]: instead of sleep here you could use `testStartTime - 2000`

**Comment by kokosing** [comment]
Merged, thanks!

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 10]
```diff
@@ -2467,6 +2470,110 @@ public void testBucketHiddenColumn()
...
+                "(0, 0), (3, 0), (6, 0), " +
+                "(1, 1), (4, 1), (7, 1), " +
+                "(2, 2), (5, 2) " +
+                " ) t(col0, col1) ";
+        assertUpdate(createTable, 8);
+        assertTrue(getQueryRunner().tableExists(getSession(), "test_file_modified_time"));
+
+        TableMetadata tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, "test_file_modified_time");
+
+        List<String> columnNames = ImmutableList.of("col0", "col1", PATH_COLUMN_NAME, FILE_SIZE_COLUMN_NAME, FILE_MODIFIED_TIME_COLUMN_NAME);
```
Comment (findepi) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java:2543]: Consider changing test method name to cover more than just "FileModifiedTimeHiddenColumn"

**Pull request merged**